### PR TITLE
Add history API support to JSDOMNodeJSEnv

### PR DESCRIPTION
**Pull request by julienrf**
This change seems to fix the problem (otherwise I was facing the following issue: https://github.com/tmpvar/jsdom/issues/1372).

### CODE REVIEW

**Comment by sjrd** [comment]
I was going to ask you to add a test for this in `NodeJSJSDOMEnvTest` ... and realize it doesn't exist :scream: 

Could you add it next to https://github.com/scala-js/scala-js/blob/master/js-envs-test-suite/src/test/scala/org/scalajs/jsenv/test/NodeJSTest.scala with a test for that support of the history API?

**Comment by julienrf** [comment]
I added a test. Without the change in the environment configuration, the test execution raises an error.

File: js-envs-test-suite/src/test/scala/org/scalajs/jsenv/test/JSDOMNodeJSEnvTest.scala [hunk 1]
```diff
@@ -0,0 +1,23 @@
...
+    """|console.log(window.location.href);
+       |window.history.pushState({}, "", "/foo");
+       |console.log(window.location.href);
+    """.stripMargin hasOutput
+    """|http://localhost/
+       |http://localhost/foo
+       |""".stripMargin
+  }
+
+}
```
Comment (sjrd) [inline js-envs-test-suite/src/test/scala/org/scalajs/jsenv/test/JSDOMNodeJSEnvTest.scala]: Missing EOL

**Comment by julienrf** [comment]
@sjrd Done.

**Comment by julienrf** [comment]
It seems that the version of jsdom that is installed on the jenkins environment is older than 9.2.0.

(cf https://github.com/tmpvar/jsdom/commit/83c410049655701c842d7e8cc3647cb0608d475d)

**Comment by sjrd** [comment]
What version is required for `changeURL`?

**Comment by julienrf** [comment]
At least 9.2.0.

**Comment by sjrd** [comment]
Jenkins, retest this please

**Pull request merged**